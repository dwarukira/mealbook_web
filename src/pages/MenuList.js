import React, { Component } from 'react'

import { connect } from 'react-redux'
import { API } from "../api"
import { MENULIST_URI } from '../constants';
import { MenuItem } from '../components/MenuItem';

import { addCart, remove, getMenu } from "../actions"

import styles from "./css/menulist.css"
import { Cart } from '../components/Cart';

class MenuList extends Component {
  
  state = {
    meals:null
  }

  addCart = (meal, quanity) => {
    this.props.dispatch(addCart(meal, quanity))
  }

  remove = (meal) => {
    this.props.dispatch(remove(meal))
  }
  componentDidMount(){
    this.props.dispatch(getMenu());
  }

  render() {
    console.log(this.props.menu.menu.menu_items);
    
    if(this.props.menu.loading){
      return (<div> Loading... </div>)
    }

    const data = this.props.menu.menu.menu_items ? this.props.menu.menu.menu_items : [] 
    const menu =  data.map(meal => (
      <MenuItem meal={meal} addCart={this.addCart} key={meal.id} remove={this.remove}/>
    ))
    return (
    <div className={`columns ${styles.columns}`}>
          <h4>Meals</h4>
          <div className={styles.menu}>
            {menu}
          </div>
        </div>


    )
  }
}


const mapStateToProps = (state) => ({
  cart:state.cart,
  menu:state.menu
})


export default connect(mapStateToProps)(MenuList)
