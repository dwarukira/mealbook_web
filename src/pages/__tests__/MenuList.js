import React from "react"
import { MenuList } from "../MenuList"
import { BrowserRouter as Router } from "react-router-dom";
import  renderer from "react-test-renderer";
describe("The MenuList Component", () => {

    describe("The MenuList ", ()=>{
        it(`Should not regress`, ()=> {
            const menu = {
                "menu":{
                    "menu_items": [
                    {
                        "meal": {
                            "id": 1,
                            "caterer": {
                                "username": "duncan",
                                "id": 1,
                                "email": "dwarukira@gmail.com"
                            },
                            "photo": "http://localhost:5000/uploads/ID_1.JPG",
                            "_links": {
                                "self": "http://localhost:5000/api/v2/meal/1",
                                "collection": "http://localhost:5000/api/v2/meals"
                            },
                            "posted_on": "2018-07-08T12:15:20.738076+00:00",
                            "name": "Cool",
                            "price": 10.0
                        },
                        "id": 9
                    },
                    {
                        "meal": {
                            "id": 2,
                            "caterer": {
                                "username": "duncan",
                                "id": 1,
                                "email": "dwarukira@gmail.com"
                            },
                            "photo": "http://localhost:5000/uploads/ID_2.JPG",
                            "_links": {
                                "self": "http://localhost:5000/api/v2/meal/2",
                                "collection": "http://localhost:5000/api/v2/meals"
                            },
                            "posted_on": "2018-07-08T12:15:49.312743+00:00",
                            "name": "test",
                            "price": 10.0
                        },
                        "id": 10
                    }]
                }
            }

            const dispatch =() =>{
                
            }
            const tree = renderer.create(
                <Router>
                    <MenuList menu={menu} dispatch={dispatch}/>
                </Router>
            )

            expect(tree.toJSON()).toMatchSnapshot()
        })
    })
   
})