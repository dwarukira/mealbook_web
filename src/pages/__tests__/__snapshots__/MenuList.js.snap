// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MenuList Component The MenuList  Should not regress 1`] = `
<div
  className="columns undefined"
>
  <h4>
    Meals
  </h4>
  <div
    className={undefined}
  >
    <div
      className="card undefined"
    >
      <img
        alt=""
        className="card-img-top"
        src="http://localhost:5000/uploads/ID_1.JPG"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Cool
        </h5>
        <p
          className="card-text"
        >
          KSH 
          10
        </p>
        <div>
          <button
            className="btn btn-primary"
            data-target="#1"
            data-toggle="modal"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="#1"
      className="modal fade"
      id={1}
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
              id="#1"
            >
              How many 
              Cool
               do you want?.
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <input
              name="quantiy"
              onChange={[Function]}
              type="number"
              value={1}
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Add
               
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card undefined"
    >
      <img
        alt=""
        className="card-img-top"
        src="http://localhost:5000/uploads/ID_2.JPG"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          test
        </h5>
        <p
          className="card-text"
        >
          KSH 
          10
        </p>
        <div>
          <button
            className="btn btn-primary"
            data-target="#2"
            data-toggle="modal"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="#2"
      className="modal fade"
      id={2}
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
              id="#2"
            >
              How many 
              test
               do you want?.
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <input
              name="quantiy"
              onChange={[Function]}
              type="number"
              value={1}
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Add
               
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
